apps/studio/src/lib/db/clients/duckdb.ts

```diff
@@ -0,0 +1,944 @@
+    const { data } = await this.driverExecuteSingle(
+      `
+        SELECT sql
+        FROM duckdb_tables
+        WHERE table_name = ?
+        AND schema_name = ?
+      `,
+      { params: [table, schema] }
+    );
+    return data[0].sql;
+  }
+
+  async getViewCreateScript(view: string, schema: string): Promise<string[]> {
+    const { data } = await this.driverExecuteSingle(
+      `
+        SELECT sql
+        FROM duckdb_views
+        WHERE view_name = ?
+        AND schema_name = ?
+      `,
+      { params: [view, schema] }
+    );
+    return data.map((row) => row.sql);
+  }
+
+  async getRoutineCreateScript(
+    _routine: string,
+    _type: string,
+    _schema: string
+  ): Promise<string[]> {
+    return [];
+  }
+
+  applyChangesSql(changes: TableChanges): string {
+    return applyChangesSql(changes, this.knex);
+  }
+
+  async applyChanges(changes: TableChanges): Promise<TableUpdateResult[]> {
+    let results = [];
+
+    await this.driverExecuteSingle("BEGIN TRANSACTION");
+
+    try {
+      if (changes.inserts) {
+        for (const command of buildInsertQueries(this.knex, changes.inserts)) {
+          await this.driverExecuteSingle(command);
+        }
+      }
+
+      if (changes.updates) {
+        results = await this.updateValues(changes.updates);
+      }
+
+      if (changes.deletes) {
+        for (const command of buildDeleteQueries(this.knex, changes.deletes)) {
+          await this.driverExecuteSingle(command);
+        }
+      }
+
+      await this.driverExecuteSingle("COMMIT");
+    } catch (ex) {
+      log.error("query exception: ", ex);
+      await this.driverExecuteSingle("ROLLBACK");
+      throw ex;
+    }
+
+    return results;
+  }
+
+  async setTableDescription(
+    table: string,
+    description: string,
+    schema: string
+  ): Promise<string> {
+    await this.driverExecuteSingle(
+      `COMMENT ON TABLE ${this.wrapIdentifier(schema)}.${this.wrapIdentifier(
+        table
+      )} IS '${description}'`
+    );
+    const result = await this.getTableProperties(table, schema);
+    return result.description;
+  }
+
+  setElementNameSql(
+    elementName: string,
+    newElementName: string,
+    typeOfElement: DatabaseElement,
+    schema?: string
+  ): string {
+    elementName = schema
+      ? `${DuckDBData.wrapIdentifier(schema)}.${DuckDBData.wrapIdentifier(
+        elementName
+      )}`
+      : DuckDBData.wrapIdentifier(elementName);
+    newElementName = DuckDBData.wrapIdentifier(newElementName);
+
+    let sql = "";
+
+    if (typeOfElement === DatabaseElement.TABLE) {
+      sql = `ALTER TABLE ${elementName} RENAME TO ${newElementName}`;
+    } else if (typeOfElement === DatabaseElement.VIEW) {
+      sql = `ALTER VIEW ${elementName} RENAME TO ${newElementName}`;
+    } else if (typeOfElement === DatabaseElement.SCHEMA) {
+      sql = `ALTER SCHEMA ${elementName} RENAME TO ${newElementName}`;
+    }
```

Comment:
Can't this just be 
```ts
sql = `ALTER ${typeOfElement} ${elementName} RENAME TO ${newElementName}`
```
?

Reply:
I think DuckDB only supports those 3 types.